pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1099,6 +1100,11 @@ func setUpstreamProtocol(cluster *apiv2.Cluster, port *model.Port) {
 			},
 		}
 	}
+
+	// Use downstream protocol. If the incoming traffic use HTTP 1.1, the
+	// upstream cluster will use HTTP 1.1, if incoming traffic use HTTP2,
+	// the upstream cluster will use HTTP2.
+	cluster.ProtocolSelection = apiv2.Cluster_USE_DOWNSTREAM_PROTOCOL
```

Comment:
so we cant do that. if a H1 client is trying to talk to h2 service, and using envoy for h1->h2 upgrade, then this will break. I think a safer option is if protocol is not http2, then we can use downstream protocol?